README.md

```diff
@@ -1,95 +1,179 @@
+memory needed for different configurations:
+
+### Llama 3.1
+
+| Model Size | Llama Variant | BF16/FP16 | FP8 | INT4(AWQ/GPTQ/bnb) |
+| :--: | :--: | :--: | :--: | :--: |
+| 1B | 3.2 | 2.5 GB | 1.25GB | 0.75GB |
+| 3B | 3.2 |6.5 GB | 3.2GB | 1.75GB |
+| 8B | 3.1 |16 GB | 8GB | 4GB |
+| 70B | 3.1 | 140 GB | 70GB | 35GB |
+|405B | 3.1 |810 GB | 405GB | 204GB |
+
+
+> [!NOTE]
+> These are estimated values and may vary based on specific
+implementation details and optimizations.
+
+Working with the capable Llama 3.1 8B models:
+
+* [Run Llama 3.1 8B in 4-bits with bitsandbytes](./4bit_bnb.ipynb)
+* [Run Llama 3.1 8B in 8-bits with bitsandbytes](./8bit_bnb.ipynb)
+* [Run Llama 3.1 8B with AWQ & fused ops](./awq.ipynb)
+
+Working on the 🐘 big Llama 3.1 405B model:
+
 * [Run Llama 3.1 405B FP8](./fp8-405B.ipynb)
 * [Run Llama 3.1 405B quantized to INT4 with AWQ](./awq_generation.py)
 * [Run Llama 3.1 405B quantized to INT4 with GPTQ](./gptq_generation.py)
-* [Run assisted decoding with Llama 405B and Llama 8B](./assisted_decoding.py)
+
+## Model Fine Tuning:
+
+It is often not enough to run inference on the model. 
+Many times, you need to fine-tune the model on some 
+custom dataset. Here are some scripts showing 
+how to fine-tune the models.
+
+Fine tune models on your custom dataset:
+* [Fine tune Llama 3.2 Vision on a custom dataset](./Llama-Vision%20FT.ipynb)
+* [Supervised Fine Tuning on Llama 3.2 Vision with TRL](./sft_vlm.py)
+* [How to fine-tune Llama 3.1 8B on consumer GPU with PEFT and QLoRA with bitsandbytes](./peft_finetuning.py)
+* [Execute a distributed fine tuning job for the Llama 3.1 405B model on a SLURM-managed computing cluster](./qlora_405B.slurm)
+
+## Assisted Decoding Techniques
+
+Do you want to use the smaller Llama 3.2 models to speedup text generation
+of bigger models? These notebooks showcase assisted decoding (speculative decoding), which gives you ~2x speedups for text generation on Llama 3.1 70B (with greedy decoding).
+
+* [Run assisted decoding with 🐘 Llama 3.1 70B and 🤏 Llama 3.2 3B](./assisted_decoding_70B_3B.ipynb)
+* [Run assisted decoding with Llama 3.1 8B and Llama 3.2 1B](./assisted_decoding_8B_1B.ipynb)
+* [Assisted Decoding with 405B model](./assisted_decoding.py)
+
+## Performance Optimization
+
+Let us optimize performace shall we?
+
 * [Accelerate your inference using torch.compile](./torch_compile.py)
 * [Accelerate your inference using torch.compile and 4-bit quantization with torchao](./torch_compile_with_torchao.ipynb)
-* Execute some Llama-generated Python code
-* Use tools with Llama!
+* [Quantize KV Cache to lower memory requirements](./quantized_cache.py)
+* [How to reuse prompts with dynamic caching](./prompt_reuse.py)
 
 ## API inference
 
 Are these models too large for you to run at home? Would you like to experiment with Llama 405B? Try out the following examples!
 
 * [Use the Inference API for PRO users](./inference-api.ipynb)
-* Use a dedicated Inference Endpoint
```

Comment:
We can actually keep this and route people to the Inference Endpoints catalog: https://ui.endpoints.huggingface.co/catalog

Reply:
Interesting -- while @osanseviero thought it would be better to remove this totally, and ask for contribution from the community.

WDYT?

Reply:
I would like it to be a community contribution -- that might be someone's first PR (that always excited me)